.github/workflows/ci.yml

```diff
@@ -9,419 +9,187 @@ on:
-        uses: actions/setup-node@v3
-        with:
-          node-version: ${{ env.PRIMARY_NODE_VERSION }}
-
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - uses: actions/cache@v3
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
-        with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
-
-      - name: Install dependencies
-        run: |
-          yarn --ignore-engines --frozen-lockfile --ignore-scripts
-          yarn check-clean-workspace-after-install
-
-      - name: Build
-        run: |
-          # Website will be built by the Netlify GitHub App
-          yarn build --exclude website
-
-      - name: Run unit tests for ast-spec
-        run: yarn test
-        working-directory: packages/ast-spec
-        env:
-          CI: true
-
-      - name: Run unit tests for typescript-estree
-        run: npx nx test @typescript-eslint/typescript-estree
-        env:
-          CI: true
-
-      - name: Run unit tests for visitor-keys
-        run: npx nx test @typescript-eslint/visitor-keys
-        env:
-          CI: true
-
-      - name: Run unit tests for scope-manager
-        run: npx nx test @typescript-eslint/scope-manager --code-coverage
-        env:
-          CI: true
-
-      - name: Run unit tests for utils
-        run: npx nx test @typescript-eslint/utils
-        env:
-          CI: true
-
-      - name: Run unit tests for type-utils
-        run: npx nx test @typescript-eslint/type-utils
-        env:
-          CI: true
-
-      - name: Run unit tests for parser
-        run: npx nx test @typescript-eslint/parser
-        env:
-          CI: true
-
-      - name: Run unit tests for eslint-plugin
-        run: npx nx test @typescript-eslint/eslint-plugin
-        env:
-          CI: true
-
-      - name: Run unit tests for eslint-plugin-tslint
-        run: npx nx test @typescript-eslint/eslint-plugin-tslint
-        env:
-          CI: true
-
-      - name: Run unit tests for eslint-plugin-internal
-        run: npx nx test @typescript-eslint/eslint-plugin-internal
-        env:
-          CI: true
-
-      - name: Publish code coverage report
-        uses: codecov/codecov-action@v3
-        with:
-          token: ${{ secrets.CODECOV_TOKEN }}
-          files: packages/**/coverage/lcov.info
-          flags: unittest
-          name: codecov
-
-  spellcheck:
-    name: Spellcheck
+jobs:
+  install:
+    name: Checkout and Install
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Use Node.js ${{ env.PRIMARY_NODE_VERSION }}
-        uses: actions/setup-node@v3
+      - name: Checkout
+        uses: actions/checkout@v3
```

Comment:
I had this originally but because `prepare-install` is a local reusable action - you have to do the checkout before GH action can reference the file :(

If we wanted - we could move the reusable actions to a separate repo entirely and then GH actions could reference them without a checkout - but I figured for now we can just eat the duplication and punt on that decision.